### PR TITLE
Jseldess/manual deployment

### PR DESCRIPTION
**Pull request by jseldess**
@mberhault, finished drafting up a page on manual deployment. It covers prerequisites, recommendations, and steps for starting up an insecure and secure multi-node cluster.

### CODE REVIEW

File: manual-deployment.md [hunk 1]
```diff
@@ -3,4 +3,164 @@ title: Manual Deployment
...
+
+On a machine that won't be part of the cluster, create the CA, node, and client certificates and keys:
+
+~~~ shell
+# Create the CA certificate and key.
+$ ./cockroach cert create-ca --ca-cert=ca.cert --ca-key=ca.key
+   
+# Create the certificate and key for nodes.
+# Be sure to include the hostnames of each machine in the cluster.
+$ ./cockroach cert create-node node1.example.com node2.example.com node3.example.com --ca-cert=ca.cert --cert=node.cert --key=node.cert
```
Comment (jseldess) [inline manual-deployment.md]: @mberhault, even after our long talk this morning, I think I got this wrong again. On this page, I'm saying to create one node cert with the hostnames of every node. Based on our talk with Ben this morning, I should recommend creating a unique certificate for each node, each time including just the specific node's hostnames (not hostnames of other nodes), is that right?

Comment (mberhault) [inline manual-deployment.md]: right. if we specify more than one hostname for a single node, it should be more along the lines of:
`./cockroach cert create-node <node1 hostname> <node1 other hostname> <node1 yet another hostname> ...`

Comment (jseldess) [inline manual-deployment.md]: OK, let me rework this a bit.

**Comment by jseldess** [comment]
@mberhault, please take another look when you have time.

File: manual-deployment.md [hunk 2]
```diff
@@ -3,4 +3,167 @@ title: Manual Deployment
...
+
+~~~ shell
+# Create the CA certificate and key.
+$ ./cockroach cert create-ca --ca-cert=ca.cert --ca-key=ca.key
+   
+# Create the node certificates and keys.
+# For each node, specify any addresses at which the node can be reached.
+$ ./cockroach cert create-node <node1-hostname> <node1-other-hostname> <node1-yet-another-hostname> --ca-cert=ca.cert --cert=node1.cert --key=node1.key
+$ ./cockroach cert create-node <node2-hostname> <node2-other-hostname> <node2-yet-another-hostname> --ca-cert=ca.cert --cert=node2.cert --key=node2.key
+$ ./cockroach cert create-node <node3-hostname> <node3-other-hostname> <node3-yet-another-hostname> --ca-cert=ca.cert --cert=node3.cert --key=node3.key
```
Comment (mberhault) [inline manual-deployment.md]: all three `create-node` lines are missing the `--ca-key=ca.key` argument.

File: manual-deployment.md [hunk 3]
```diff
@@ -3,4 +3,167 @@ title: Manual Deployment
...
+~~~
+
+Store the CA key somewhere safe and keep a backup; if you lose it, you will not be able to add new nodes or clients to your cluster.
+
+### 2. Set up the first node
+
+Copy the `cockroach` binary, CA certificate, and node 1 certificate and key to the first machine and then start the node:
+
+~~~ shell
+$ ./cockroach start --ca-cert=certs/ca.cert --cert=certs/node1.cert --key=node1.key --host=<node1-hostname>
```
Comment (mberhault) [inline manual-deployment.md]: the cert paths are inconsistent here. let's drop `certs/` entirely.
eg:
`$ ./cockroach start --ca-cert=ca.cert --cert=node1.cert --key=node1.key --host=<node1-hostname>`

File: manual-deployment.md [hunk 4]
```diff
@@ -3,4 +3,167 @@ title: Manual Deployment
...
+~~~
+
+This command specifies the location of certificates and the address at which other nodes can reach it. Otherwise, it uses all available defaults. For example, the node stores data in the `cockroach-data` directory, listens for internal and client communication on port 26257, and listens for HTTP requests from the Admin UI on port 8080. To set these options manually, see [Start a Node](start-a-node.html). 
+
+### 3. Set up the second node
+
+Copy the `cockroach` binary, CA certificate, and node 2 certificate and key to the second machine and then start the node:
+
+~~~ shell
+./cockroach start --ca-cert=certs/ca.cert --cert=certs/node2.cert --key=node2.key --host=<node2-hostname> --join=<node1-hostname>:26257
```
Comment (mberhault) [inline manual-deployment.md]: ditto: let's drop `certs/`

File: manual-deployment.md [hunk 5]
```diff
@@ -3,4 +3,167 @@ title: Manual Deployment
...
+### 4. Set up additional nodes
+
+Repeat step 3 for each additional node. 
+
+### 5. Use the built-in SQL client
+
+You can run the built-in SQL client from any machine with the `cockroach` binary, CA cert, client certificate, and client key. Make sure the machine you want to use has these files and then start the client:  
+
+~~~ shell
+$ ./cockroach sql --url=postgresql://root@<node1-hostname>:26257?sslcert=certs/root.crt&sslkey=certs/root.key&sslmode=verify-full&sslrootcert=certs/ca.crt
```
Comment (mberhault) [inline manual-deployment.md]: also drop `certs/` here.
furthermore, this command specified `.crt` as the extention, but all previous steps use `.cert`

We should probably have a `/` before the `?` in the url

File: manual-deployment.md [hunk 6]
```diff
@@ -3,4 +3,167 @@ title: Manual Deployment
...
+### 3. Set up additional nodes
+
+Repeat step 2 for each additional node.
+
+### 4. Use the Built-in SQL Client
+
+Start the built-in SQL client from any machine with the `cockroach` binary. This could be one of the node machines or a different machine. 
+
+~~~ shell
+$ ./cockroach sql --insecure --url=postgresql://root@<node-hostname>:26257?sslmode=disable
```
Comment (mberhault) [inline manual-deployment.md]: let's put a `/` before the `?` in the URL

File: manual-deployment.md (hunk 2 above)
Comment (jseldess) [inline manual-deployment.md]: Fixed with last commit.

File: manual-deployment.md (hunk 3 above)
Comment (jseldess) [inline manual-deployment.md]: Fixed with last commit.

File: manual-deployment.md (hunk 4 above)
Comment (jseldess) [inline manual-deployment.md]: Fixed with last commit.

File: manual-deployment.md (hunk 5 above)
Comment (jseldess) [inline manual-deployment.md]: Fixed with last commit.

File: manual-deployment.md (hunk 6 above)
Comment (jseldess) [inline manual-deployment.md]: Fixed with last commit.

**Comment by mberhault** [comment]
nice! LGTM

**Comment by jseldess** [comment]
@mberhault, one more commit: Added steps for connecting with client drivers and monitoring with the admin ui. Still look ok?

**Comment by mberhault** [comment]
Yup. LGTM

**Pull request merged**